buildSrc/src/main/kotlin/non-multiplatform.publish-conventions.gradle.kts

```diff
@@ -0,0 +1,31 @@
+plugins {
+//     Regular java modules need 'java-library' plugin for proper publication
+    `java-library`
+    id("publish-conventions")
+}
+
+// MPP projects pack their sources automatically, java libraries need to explicitly pack them
+val sourcesJar = tasks.register<Jar>("sourcesJar") {
+    archiveClassifier.set("sources")
+    from("src/main/kotlin")
+}
+
+publishing {
+    // Configure java publications for non-MPP projects
+    publications {
+        if (project.name == "atomicfu-gradle-plugin") {
+            create<MavenPublication>("pluginMaven") {
+                artifact(sourcesJar)
+            }
+        } else {
+            create<MavenPublication>("maven") {
+                from(components["java"])
+                artifact(sourcesJar)
+                if (project.name.endsWith("-maven-plugin")) {
+                    pom.packaging = "maven-plugin"
+                }
```

Comment:
It'd make sense to move this subproject-specific logic into the subproject's build.gradle. Convention plugins should (ideally) be unopinionated, and avoid having subproject specific logic.

Reply:
Moved + rewrote `build.gradle` to `build.gradle.kts` in `kotlinx-atomicfu.atomicfu-maven-plugin`, thus could you please have a look on this change too